lib/tokenize.cpp

```diff
@@ -8463,6 +8463,16 @@ void Tokenizer::findGarbageCode() const
     // Operators without operands..
     const Token *templateEndToken = nullptr;
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
+        // Skip preprocessor directives
```

Comment:
preprocessor directives are supposed to be removed before this code is reached.

Reply:
It seems some test have `# line 1` in the tokenlist in our test suite. This is to ensure we dont throw syntax errors for such constructs.

Reply:
thanks. hmm.. weird test. it seems we need some hack(s) in the code to make that work. my spontaneous reaction is that we should fix that test. you do not need to fix it in this PR.

Reply:
It seems like the test is there to make sure it works with the preprocessed output from the compiler, which can have `#line` directives in it. The test is not wrong, although its strange this is tested with uninitialized variables.

Reply:
ok I did not understand that part. Yes then it is strange location! It should be tested in `simplecpp` then. As far as I understand the `simplecpp::preprocess` does not add any directives in its output so that is where this is removed.

Reply:
The checker does not care if it's a header file or source file so I see no point to keep these tests. I remove these.. and then I will add some proper simplecpp tests asap.